### PR TITLE
rofisim: Compiling without gazebo

### LINKED ISSUE #76: Compiling without Gazebo
- [x] Add switch to `rcfg` to allow compilation without the need to install gazebo libraries
- [x] Remove global linking of gazebo libraries and directories

### PR DESCRIPTION
**Pull request by Ondraceq**
Resolves #76

### CODE REVIEW

**Comment by Ondraceq** [comment]
TODO checks:
 - [ ] build with gazebo (removed global linking)
 - [ ] build docs without gazebo installed
 - [ ] build world-creator (the armadillo dependency in cmake was removed - use ${ARMADILLO_LIBRARIES} if needed)

File: softwareComponents/gazebosim/common/CMakeLists.txt [hunk 1]
```diff
@@ -2,4 +2,5 @@ cmake_minimum_required(VERSION 3.11)
 
 add_library(rofiPluginCommon INTERFACE)
 target_link_libraries(rofiPluginCommon INTERFACE ${GAZEBO_LIBRARIES} ${Boost_LIBRARIES} rofisimMessages)
+target_include_directories(rofiPluginCommon INTERFACE ${GAZEBO_INCLUDE_DIRS})
```
Comment (yaqwsx) [inline softwareComponents/gazebosim/common/CMakeLists.txt:5]: Please consider merging these two lines (5 & 6)

**Review by yaqwsx — CHANGES_REQUESTED** [review]
I am not sure if I follow what "build world-creator (the armadillo dependency in cmake was removed - use ${ARMADILLO_LIBRARIES} if needed)" means.

**Comment by Ondraceq** [comment]
> 
> I am not sure if I follow what "build world-creator (the armadillo dependency in cmake was removed - use ${ARMADILLO_LIBRARIES} if needed)" means.

It was just a reminder to check if the tool compiles, because I didn't have the library available at the moment

**Review by yaqwsx — APPROVED** [review]

**Pull request merged**